### ISSUE TITLE
Error while using Convert_to_pb to convert yolov3 tiny weights into tensorflow

### ISSUE THREAD

**Issue by srikar242** [pythonlessons/TensorFlow-2.x-YOLOv3#63]
Hi @pythonlessons 
    I have trained a custom tiny-yolov3-prn model using AlexCarey Darknet. While using Convert_to_pb script to convert darknet weights into tensorflow, Iam facing the below error:

 ```
     
  Traceback (most recent call last):
  File "Convert_to_pb.py", line 35, in <module>
    yolo.load_weights(YOLO_CUSTOM_WEIGHTS) # use custom weights
  File "/home/srikar/anaconda3/envs/yolov3/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 2172, in load_weights
    if _is_hdf5_filepath(filepath):
  File "/home/srikar/anaconda3/envs/yolov3/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 2785, in _is_hdf5_filepath
    return (filepath.endswith('.h5') or filepath.endswith('.keras') or
AttributeError: 'bool' object has no attribute 'endswith'
```

Any help in this regard is much appreciated.

**Comment by bhack** [comment]
Please share a very, very minimal standalone code example (or a Colab) that we just could copy, paste and run.

**Comment by srikar242** [comment]
Thanks for your reply @pythonlessons . I am just using the script Convert_to_pb.py which is inside tools folder in this repo. I have followed all the steps required to convert custom darknet weights. I have changed config.py to convert custom weights. And then,I am using the conversion script. I haven't changed anything in that script. Which code should I share?

**Comment by bhack** [comment]
I config you need to set `YOLO_CUSTOM_WEIGHTS=<your_custom_weights_path>`

**Comment by srikar242** [comment]
Yeah.I did that.Now I am getting different error.

Traceback (most recent call last):
  File "Convert_to_pb.py", line 35, in <module>
    yolo.load_weights(YOLO_CUSTOM_WEIGHTS) # use custom weights
  File "/home/srikar/anaconda3/envs/yolov3/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 2204, in load_weights
    with h5py.File(filepath, 'r') as f:
  File "/home/srikar/anaconda3/envs/yolov3/lib/python3.7/site-packages/h5py/_hl/files.py", line 408, in __init__
    swmr=swmr)
  File "/home/srikar/anaconda3/envs/yolov3/lib/python3.7/site-packages/h5py/_hl/files.py", line 173, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 88, in h5py.h5f.open
OSError: Unable to open file (file signature not found)

My file is not corrupted.
 My config file:
# YOLO options
YOLO_TYPE                   = "yolov3" # yolov4 or yolov3
YOLO_FRAMEWORK              = "tf" # "tf" or "trt"
YOLO_V3_WEIGHTS             = "model_data/yolov3.weights"
YOLO_V4_WEIGHTS             = "model_data/yolov4.weights"
YOLO_V3_TINY_WEIGHTS        = "model_data/yolov3-tiny.weights"
YOLO_V4_TINY_WEIGHTS        = "model_data/yolov4-tiny.weights"
YOLO_TRT_QUANTIZE_MODE      = "INT8" # INT8, FP16, FP32
YOLO_CUSTOM_WEIGHTS         = "model_data/yolov3-tiny-prn-obj_5000.weights" # "checkpoints/yolov3_custom" # used in evaluate_mAP.py and custom model detection, if not using leave False
                            # YOLO_CUSTOM_WEIGHTS also used with TensorRT and custom model detection
YOLO_COCO_CLASSES           = "model_data/obj.names"
YOLO_STRIDES                = [8, 16, 32]
YOLO_IOU_LOSS_THRESH        = 0.5
YOLO_ANCHOR_PER_SCALE       = 3
YOLO_MAX_BBOX_PER_SCALE     = 100
YOLO_INPUT_SIZE             = 416
if YOLO_TYPE                == "yolov4":
    YOLO_ANCHORS            = [[[12,  16], [19,   36], [40,   28]],
                               [[36,  75], [76,   55], [72,  146]],
                               [[142,110], [192, 243], [459, 401]]]
if YOLO_TYPE                == "yolov3":
    YOLO_ANCHORS            = [[[10,  13], [16,   30], [33,   23]],
                               [[30,  61], [62,   45], [59,  119]],
                               [[116, 90], [156, 198], [373, 326]]]
# Train options
TRAIN_YOLO_TINY             = True
TRAIN_SAVE_BEST_ONLY        = True # saves only best model according validation loss (True recommended)
TRAIN_SAVE_CHECKPOINT       = False # saves all best validated checkpoints in training process (may require a lot disk space) (False recommended)
TRAIN_CLASSES               = "model_data/obj.names"
TRAIN_ANNOT_PATH            = "mnist/mnist_train.txt"
TRAIN_LOGDIR                = "log"
TRAIN_CHECKPOINTS_FOLDER    = "checkpoints"
TRAIN_MODEL_NAME            = f"{YOLO_TYPE}_custom"
TRAIN_LOAD_IMAGES_TO_RAM    = True # With True faster training, but need more RAM
TRAIN_BATCH_SIZE            = 4
TRAIN_INPUT_SIZE            = 416
TRAIN_DATA_AUG              = True
TRAIN_TRANSFER              = True
TRAIN_FROM_CHECKPOINT       = False # "checkpoints/yolov3_custom"
TRAIN_LR_INIT               = 1e-4
TRAIN_LR_END                = 1e-6
TRAIN_WARMUP_EPOCHS         = 2
TRAIN_EPOCHS                = 100

# TEST options
TEST_ANNOT_PATH             = "mnist/mnist_test.txt"
TEST_BATCH_SIZE             = 4
TEST_INPUT_SIZE             = 416
TEST_DATA_AUG               = False
TEST_DECTECTED_IMAGE_PATH   = ""
TEST_SCORE_THRESHOLD        = 0.3
TEST_IOU_THRESHOLD          = 0.45


#YOLOv3-TINY and YOLOv4-TINY WORKAROUND
if TRAIN_YOLO_TINY:
    YOLO_STRIDES            = [16, 32, 64]    
    YOLO_ANCHORS            = [[[10,  14], [23,   27], [37,   58]],
                               [[81,  82], [135, 169], [344, 319]],
                               [[0,    0], [0,     0], [0,     0]]]

**Comment by pythonlessons** [comment]
Hi,
Because you trained custom model with custom classes count with AlexCarey it wont work like that. 
Right now code is written that you can convert standard yolo model or custom tf model.
Because you have custom model not on tensorflow, it won't work with `yolo.load_weights(YOLO_CUSTOM_WEIGHTS)`, you should use `load_yolo_weights`

**Comment by srikar242** [comment]
Thanks @pythonlessons for your reply. I changed the function from yolo.load_weights to load_yolo_weights .However, I am getting  reshape error. I understand that it is because of indifference between input and output. I have checked my names file and it is alright.

Traceback (most recent call last):
  File "Convert_to_pb.py", line 35, in <module>
    load_yolo_weights(yolo, YOLO_CUSTOM_WEIGHTS) # use custom weights
  File "/home/srikar/Downloads/TensorFlow-2.x-YOLOv3-master/yolov3/utils.py", line 66, in load_yolo_weights
    conv_weights = conv_weights.reshape(conv_shape).transpose([2, 3, 1, 0])
ValueError: cannot reshape array of size 3285758 into shape (1024,512,3,3)
